packages/assert/src/assert.js

```diff
@@ -57,6 +57,21 @@ export { quote as q };
 
 export { makeAssert };
 
+/**
+ * @template T
+ * @param {T | null | undefined} val
+ * @param {string} [optDetails]
+ * @returns {T}
+ */
```

Comment:
Is this adequate for TS to infer that the returned `T` is not `null` or `undefined`? Do we need a TS `asserts` or something in the return?

Reply:
It works because if it were `null` or `undefined` then by line 62 it wouldn't be `T`.